backend/clubs/serializers.py

```diff
@@ -2424,6 +2425,51 @@ class Meta:
         fields = ("text", "multiple_choice", "question_type", "question")
 
 
+class ApplicationExtensionSerializer(serializers.ModelSerializer):
+    first_name = serializers.CharField(source="user.first_name", read_only=True)
+    last_name = serializers.CharField(source="user.last_name", read_only=True)
+    username = serializers.CharField(source="user.username", read_only=False)
+    graduation_year = serializers.CharField(
+        source="user.profile.graduation_year", read_only=True
+    )
+
+    class Meta:
+        model = ApplicationExtension
+        fields = (
+            "id",
+            "username",
+            "first_name",
+            "last_name",
+            "graduation_year",
+            "end_time",
+        )
+
+    def validate_username(self, value):
+        user = get_user_model().objects.filter(username=value).first()
+        if not user:
+            raise serializers.ValidationError("Please provide a valid username!")
+        return value
+
+    def create(self, validated_data):
+        username = validated_data.get("user").pop("username")
+        validated_data["user"] = get_user_model().objects.get(username=username)
+
+        application_pk = self.context["view"].kwargs.get("application_pk")
+        validated_data["application"] = ClubApplication.objects.filter(
```

Comment:
Thoughts on validating `application_pk`?

Reply:
And also the `end_time` should be greater than the existing end time...then you can get rid of the `max()` stuff below

Reply:
Yep sorry, meant to open this as a draft PR as I was gonna update validation logic. Also, I added `max()` in case the club updates their overall application deadline after the individual extension is issued